docs(contributors): explain the usage of PR labels (#3269)
process-analytics/bpmn-visualization-js: docs/contributors/maintainers.md

> - It has been approved by at least two maintainers. If it was a maintainer who opened the PR, only one approval is needed.

Reviewer: **suggestion:** 
```suggestion
- It has at least two maintainer approvals (one if it was opened by a maintainer).
```

- It has no requested changes.
- It is up-to-date with the `master` branch.

> - The related labels are correctly set because the GitHub release will include an auto-generated release notes which is based on the labels of the merged Pull Requests.

Reviewer: **suggestion:** 
```suggestion
- Labels are correctly set for auto-generated release notes.
```

  This ensures that development can continue without impacting the writing of the content of the `In Progress` release. That way,
  if a PR is merged, `release-drafter` will update the `Next` draft release keeping the `In Progress` release untouched.

> The GitHub release will include an auto-generated release notes which is based on the labels of the merged Pull Requests:
>   - Ensure that all merged PR included in the release are labelled. You can find the [unlabeled PR](https://github.com/process-analytics/bpmn-visualization-js/pulls?q=is%3Apr+sort%3Aupdated-desc+no%3Alabel+is%3Amerged) to easily labeled them.
>   - Ensure that all PR related to the bump of production dependencies (mxGraph, ...) are not labelled with `skip-changelog`, otherwise they are not included in the changelog.
>   Such PR can be found with the following query: [bump prod dependency PR with skip-changelog label](https://github.com/process-analytics/bpmn-visualization-js/pulls?q=is%3Apr+is%3Amerged+%22chore%28deps%29%22+in%3Atitle+NOT+%22chore%28deps-dev%29%22+in%3Atitle+label%3Askip-changelog+)

Reviewer: **suggestion:** 
```suggestion
The GitHub release includes an auto-generated release notes based on merged PR labels. To ensure this works:
  -  Label all merged PRs in the release. Check [unlabeled PRs](https://github.com/process-analytics/bpmn-visualization-js/pulls?q=is%3Apr+sort%3Aupdated-desc+no%3Alabel+is%3Amerged) to add missing labels.
  - Ensure that all PRs related to bump production dependencies (e.g., mxGraph) do not have the `skip-changelog` label. Otherwise, they are not included in the changelog.
  Find such PRs here: [bump prod dependency PR with skip-changelog label](https://github.com/process-analytics/bpmn-visualization-js/pulls?q=is%3Apr+is%3Amerged+%22chore%28deps%29%22+in%3Atitle+NOT+%22chore%28deps-dev%29%22+in%3Atitle+label%3Askip-changelog+).
```

> **Note**: It is currently not possible to make dependabot apply different labels to production and development dependency update PRs. So, that's why we have to check the labels manually.
> For more details, see the documentation about the [dependabot labels configuration](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#labels--).

Reviewer: **suggestion:** 
```suggestion
**Note**: Currently, dependabot can't apply different labels to production and development dependency update PRs, so labels must be checked manually.
For more details, see the [dependabot labels documentation](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#labels--).
```
